src/librustdoc/core.rs

```diff
@@ -416,10 +447,16 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
             let mut global_ctxt = abort_on_err(queries.global_ctxt(), sess).take();
 
             global_ctxt.enter(|tcx| {
-                tcx.analysis(LOCAL_CRATE).ok();
-
-                // Abort if there were any errors so far
-                sess.abort_if_errors();
+                // Some queries require that they only run on valid types:
+                // https://github.com/rust-lang/rust/pull/73566#issuecomment-656954425
+                // Therefore typecheck this crate before running lints.
```

Comment:
Not sure if this phrasing is ideal. I'd say that "certain queries (in)directly used by `rustdoc` assume some checks were performed elsewhere, so we do as much as we can without type-checking bodies".

Reply:
I cleaned up the wording a little, how does it look now?